### PR TITLE
Require CMake 3.10 and Check SSE support directly

### PR DESCRIPTION
**Pull request by sloretz**
Require CMake 3.10 and check if SSE/SSE2 are supported directly.
~Check if SSE/SSE2 are supported if the CMake version is new enough to do so.~

I think this would fix a build failure on armhf and aarch64:

* http://build.ros.org/view/Nbin_ufhf_uFhf/job/Nbin_ufhf_uFhf__fcl__ubuntu_focal_armhf__binary/13/console
* http://build.ros.org/view/Nbin_ufv8_uFv8/job/Nbin_ufv8_uFv8__fcl__ubuntu_focal_arm64__binary/13/console

```
16:33:57 c++: error: unrecognized command line option ‘-mfpmath=sse’
16:33:57 c++: error: unrecognized command line option ‘-msse’
16:33:57 c++: error: unrecognized command line option ‘-msse2’
16:33:57 c++: error: unrecognized command line option ‘-msse3’
16:33:57 c++: error: unrecognized command line option ‘-mssse3’
```

@rhaschke Mind taking a look?

---
This change is [Reviewable](https://reviewable.io/reviews/flexible-collision-library/fcl/506)

### CODE REVIEW

**Review by rhaschke — APPROVED** [review]
Looks reasonable to me. Thanks, @sloretz for handling this.

**Review by SeanCurtis-TRI — APPROVED** [review]
@jamiesnape can you give this your blessing? Seems reasonable to me, but I defer to your expertise.

> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/flexible-collision-library/fcl/506#-:-MK_Nfu89X4z5TDyqr88:bg8wy50)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by SeanCurtis-TRI — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/flexible-collision-library/fcl/506)* status: all files reviewed, 1 unresolved discussion (waiting on @sloretz)

*[a discussion](https://reviewable.io/reviews/flexible-collision-library/fcl/506#-MK_pl52EQj9kgKyAPr4:-MK_pl52EQj9kgKyAPr5:bybr0d8) (no related file):*
Blocking while we wait for @jamiesnape 

---

**Review by SeanCurtis-TRI — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/flexible-collision-library/fcl/506)* status: all files reviewed, 1 unresolved discussion (waiting on @jamiesnape and @sloretz)

*[a discussion](https://reviewable.io/reviews/flexible-collision-library/fcl/506#-MK_pl52EQj9kgKyAPr4:-MK_pqnG7Nv6R93arw5F:bebkkm5) (no related file):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

Blocking while we wait for @jamiesnape 
</blockquote></details>

+@jamiesnape 

---

**Review by jamiesnape — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/flexible-collision-library/fcl/506)* status: all files reviewed, 2 unresolved discussions (waiting on @sloretz)

___
*[CMakeLists.txt, line 91 at r1](https://reviewable.io/reviews/flexible-collision-library/fcl/506#-MKbNcjz1qbsEWSFPpEY:-MKbNcjz1qbsEWSFPpEZ:b-t428ya) ([raw file](https://github.com/flexible-collision-library/fcl/blob/f0d1b0041a9376b9ab57b80d37df82ed5bc8808c/CMakeLists.txt#L91)):*
> ```CMake
> if(CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
>   set(FCL_TARGET_SUPPORT_X64_SSE OFF)
> elseif(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.10.0")
> ```

According to the minimum required versions of CMake for this project if(... VERSION_GREATER_EQUAL ...) is not supported. Dropping support for old versions of CMake is probably an issue elsewhere.

**Review by SeanCurtis-TRI — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/flexible-collision-library/fcl/506)* status: all files reviewed, 2 unresolved discussions (waiting on @jamiesnape and @sloretz)

___
*[CMakeLists.txt, line 91 at r1](https://reviewable.io/reviews/flexible-collision-library/fcl/506#-MKbNcjz1qbsEWSFPpEY:-MKeBOCy6ng6CLUOYD8D:bpxpkut) ([raw file](https://github.com/flexible-collision-library/fcl/blob/f0d1b0041a9376b9ab57b80d37df82ed5bc8808c/CMakeLists.txt#L91)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

According to the minimum required versions of CMake for this project if(... VERSION_GREATER_EQUAL ...) is not supported. Dropping support for old versions of CMake is probably an issue elsewhere.
</blockquote></details>

What is the minimum version of cmake that supports `VERSION_GREATER_EQUAL`? We recently bumped all the CI requirements to less antiquated versions of Mac OS and Ubuntu, could we simply do the same with CMake? Something that could be advanced here enough to provide the missing functionality, but doesn't require further changes to the current spelling (deferring strategic changes to the build infrastructure to subsequent PRs)?

File: CMakeLists.txt
Comment (rhaschke) [inline CMakeLists.txt:100]: `VERSION_GREATER_EQUAL` was introduced in cmake 3.7 only. But we can easily invert the logics and use `VERSION_LESS` that is [available in cmake 2.8](https://cmake.org/cmake/help/v2.8.12/cmake.html) as well:
```suggestion
elseif(${CMAKE_VERSION} VERSION_LESS "3.10.0")
  set(FCL_TARGET_SUPPORT_X64_SSE ON)
else() # Decide based on actual host features (only supported since cmake 3.10)
  cmake_host_system_information(RESULT _has_sse QUERY HAS_SSE)
  cmake_host_system_information(RESULT _has_sse2 QUERY HAS_SSE2)
  if(_has_sse AND _has_sse2)
    set(FCL_TARGET_SUPPORT_X64_SSE ON)
  else()
    set(FCL_TARGET_SUPPORT_X64_SSE OFF)
  endif()
```

**Review by jamiesnape — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/flexible-collision-library/fcl/506)* status: all files reviewed, 3 unresolved discussions (waiting on @SeanCurtis-TRI and @sloretz)

___
*[CMakeLists.txt, line 91 at r1](https://reviewable.io/reviews/flexible-collision-library/fcl/506#-MKbNcjz1qbsEWSFPpEY:-MKfZrbp2si61WB_8o-h:b-lbp5fb) ([raw file](https://github.com/flexible-collision-library/fcl/blob/f0d1b0041a9376b9ab57b80d37df82ed5bc8808c/CMakeLists.txt#L91)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

What is the minimum version of cmake that supports `VERSION_GREATER_EQUAL`? We recently bumped all the CI requirements to less antiquated versions of Mac OS and Ubuntu, could we simply do the same with CMake? Something that could be advanced here enough to provide the missing functionality, but doesn't require further changes to the current spelling (deferring strategic changes to the build infrastructure to subsequent PRs)? 
</blockquote></details>

If the oldest platform is Bionic, then we can remove everything at https://github.com/flexible-collision-library/fcl/blob/b611d50e1246fad8dcb445117f84259792a37e1a/CMakeLists.txt#L34-L40 and replace it with `cmake_minimum_required(VERSION 3.10)` and remove the `if()` statements here.

**Review by SeanCurtis-TRI — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/flexible-collision-library/fcl/506)* status: all files reviewed, 3 unresolved discussions (waiting on @jamiesnape and @sloretz)

___
*[CMakeLists.txt, line 91 at r1](https://reviewable.io/reviews/flexible-collision-library/fcl/506#-MKbNcjz1qbsEWSFPpEY:-MKfgn-j5TOwbzThZwox:b-11syqa) ([raw file](https://github.com/flexible-collision-library/fcl/blob/f0d1b0041a9376b9ab57b80d37df82ed5bc8808c/CMakeLists.txt#L91)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

If the oldest platform is Bionic, then we can remove everything at https://github.com/flexible-collision-library/fcl/blob/b611d50e1246fad8dcb445117f84259792a37e1a/CMakeLists.txt#L34-L40 and replace it with `cmake_minimum_required(VERSION 3.10)` and remove the `if()` statements here.
</blockquote></details>

We've bumped CI to bionic and I don't see us going backwards at all. I'm game for bumping the CMake version and simplifying the code.

**Comment by sloretz** [comment]
> If the oldest platform is Bionic, then we can remove everything at [...] and replace it with cmake_minimum_required(VERSION 3.10) and remove the if() statements here.

> We've bumped CI to bionic and I don't see us going backwards at all. I'm game for bumping the CMake version and simplifying the code.

@SeanCurtis-TRI @jamiesnape It sounds like the minimum CMake version is going to be bumped to 3.10. Is that something you'd like to see in this PR, or is another one coming? I assumed the latter and removed the CMake version check in 5a87fff.

**Comment by jamiesnape** [comment]
I think adding a separate commit before the current one that makes the change to `cmake_minimum_required(VERSION 3.10)` would be order. Either way that needs to happen before the current commit once merged into master.

**Review by SeanCurtis-TRI — COMMENTED** [review]
Agreed.

> *[Reviewable](https://reviewable.io/reviews/flexible-collision-library/fcl/506#-:-MKu8sk_AwI-gPXPCJIY:b-pbdqz3)* status: 0 of 1 files reviewed, 3 unresolved discussions (waiting on @jamiesnape, @SeanCurtis-TRI, and @sloretz)

**Comment by sloretz** [comment]
> I think adding a separate commit before the current one that makes the change to cmake_minimum_required(VERSION 3.10) would be in order. Either way that needs to happen before the current commit once merged into master.

The first commit 7c1dd82 now bumps the CMake requirement to 3.10, and the second commit f4e5530 checks SSE support

**Review by jamiesnape — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2.
> Dismissed @rhaschke from a discussion.
> *[Reviewable](https://reviewable.io/reviews/flexible-collision-library/fcl/506#-:-MLNkCs1AaZsXcLMVuud:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @sloretz)

**Review by SeanCurtis-TRI — APPROVED** [review]
I'll use my super powers to merge as two commits.

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/flexible-collision-library/fcl/506#-:-MLOuyLyCIkLHcPdjJl0:b-7hcglm)* status: :shipit: complete! all files reviewed, all discussions resolved

**Comment by SeanCurtis-TRI** [comment]
Thanks for the PR!

**Pull request merged**
